tss-esapi/src/structures/pcr/slot_collection.rs

```diff
@@ -0,0 +1,240 @@
+// Copyright 2022 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    structures::{PcrSelectSize, PcrSlot},
+    tss2_esys::TPM2_PCR_SELECT_MAX,
+    Error, Result, WrapperErrorKind,
+};
+use enumflags2::BitFlags;
+use log::error;
+use std::convert::TryFrom;
+
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub struct PcrSlotCollection {
```

Comment:
I was actually thinking of doing that. But as you mentioned the PcrSelect is not a part of those two types according to the specification(even though I think that just an error by those who wrote it). So I wanted to be true to the specification with the types but still make all the conversions in one place to avoid code duplication.

Reply:
Yeah, makes sense. Do you think we even need to make this struct public, though? Does it get used anywhere as an input/output?

Reply:
Ah, I only just noticed that at `structures` level you made it `pub(crate)`. All good!